src/theme/theme.ts

```diff
@@ -0,0 +1,67 @@
+import {makeAutoObservable, runInAction} from 'mobx';
+import {makePersistable} from 'mobx-persist-store';
+import {AppState, Appearance, StatusBar} from 'react-native';
+
+// TODO Move app status into separate store
+import {AppStatus, getAppStatus} from '@app/contexts';
+import {storage} from '@app/services/mmkv';
+
+import {AppTheme} from './types';
+
+class Theme {
+  _systemTheme = Appearance.getColorScheme() as AppTheme;
+  _currentTheme: AppTheme =
+    (Appearance.getColorScheme() as AppTheme) ?? AppTheme.light;
+
+  constructor() {
+    makePersistable(this, {
+      name: this.constructor.name,
+      properties: ['_currentTheme'],
+      storage: storage,
+    });
+
+    AppState.addEventListener('change', this.onChangeThemeListener);
+    Appearance.addChangeListener(this.onChangeThemeListener);
+
+    makeAutoObservable(this);
+  }
+
+  onChangeThemeListener = () => {
+    const systemColorScheme = Appearance.getColorScheme() as AppTheme;
+
+    if (getAppStatus() === AppStatus.inactive) {
+      return;
+    }
+
+    runInAction(() => {
+      this._systemTheme = systemColorScheme;
+      this._currentTheme =
+        this._currentTheme === AppTheme.system
+          ? this._systemTheme ?? AppTheme.light
+          : this._currentTheme;
```

Comment:
if **currentTheme** equals **system**, then you override **currentTheme**.
I think it's a bug, because **currentTheme** ceases to be system which user set in settings